web/modules/custom/pulses_exchange_rate/src/CurrencyDataProvider.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Drupal\pulses_exchange_rate;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\pulses_exchange_rate\Form\ExchangeApiKey;
+use GuzzleHttp\ClientInterface;
+
+/**
+ * Object for get data from api.
+ */
+class CurrencyDataProvider implements CurrencyDataProviderInterface {
+
+  /**
+   * Client instance.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  public $client;
+
+  /**
+   * Http header.
+   *
+   * @var object
+   */
+  public $header;
+
+  /**
+   * Final array with different nested.
+   *
+   * @var array
+   */
+  public $data = [];
+
+  /**
+   * Instance of ConfigFactoryInterface.
+   *
+   * @var \Drupal\pulses_exchange_rate\CurrencyDataProvider
+   */
+  public $configFactory;
+
+  /**
+   * Instance of Client class.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   Client object.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   ConfigFactory object.
+   */
+  public function __construct(ClientInterface $http_client, ConfigFactoryInterface $config_factory) {
+    $this->client = $http_client;
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * Request data by URL & write value to object variable (header).
+   *
+   * Sort array with data after decode & write value to object variable (data).
+   *
+   * @param string $url
+   *   Url of API page.
+   * @param string $api_key
+   *   Single parameter for API Url.
+   * @param bool $nested
+   *   Describe how to sort array with data after decode.
+   */
+  public function getResponse($url, $api_key, $nested) {
```

Comment:
Only one current function must be `public`, the others must be `private` or `protected` because they are just additional functions of the current class.